app/scripts/background.js

```diff
@@ -175,6 +175,23 @@ if (globalThis.stateHooks.onInstalledListener) {
   });
 }
 
+/**
+ * Trigger actions that should happen only when an update is available
+ */
+function onUpdateAvailable() {
+  if (controller) {
+    log.debug('An update is available');
+    controller.appStateController.setIsUpdateAvailable(true);
+    return;
+  }
+  setTimeout(() => {
+    // If the controller is not set yet, we wait and try again
+    onUpdateAvailable();
+  }, 500);
```

Comment:
You should `await isInitialized` here instead of timeouts. Timeouts are fine for retries against remote code or code we don't have control over, not for code that we can easily hook into.